### PR TITLE
Fix check_format.sh to work on Mac OSX and on Travis (YAPF)

### PR DESCRIPTION
**Pull request by s-mandra**

### CODE REVIEW

**Comment by s-mandra** [comment]
For consistency, `check_format.sh` is now run inside a docker containers (slightly different versions of `yapf` and `clang-format` give different formats. Also, styles are now saved in `.style.yapf` and `.clang-format` in the root folder.

**Comment by s-mandra** [comment]
Fixes #206

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
   - docker
 
-addons:
-  apt:
-    packages:
-    - clang-format
-
 before_script:
-  - bash scripts/check_format.sh
+  - docker run -ti --rm -v $PWD:/qflex:ro debian:bullseye /bin/bash -c "apt update && apt install -y yapf3 clang-format && bash /qflex/scripts/check_format.sh"
```
Comment (95-martin-orion) [inline .travis.yml]: This check appears to work as intended, but why do we use `debian:bullseye` (instead of e.g. `alpine`)?

File: scripts/check_format.sh [hunk 2]
```diff
@@ -5,6 +5,20 @@
 # be printed as output.
 
 ROOT_DIR="$(realpath $(dirname $0))/../"
+if which yapf3 > /dev/null; then
```
Comment (95-martin-orion) [inline scripts/check_format.sh]: I'll defer to @alankao64 for checking this on Mac OSX, as I no longer have easy access to a device running it.

File: scripts/check_format.sh [hunk 3]
```diff
@@ -45,7 +59,7 @@ for filename in $(find_cmd ${ROOT_DIR}/ -type f -iname "*.h" -or -iname "*.cpp")
   filename=$(realpath $(dirname $filename))/$(basename $filename)
   echo "Checking: $filename" >&2
   # ...check if there are any changes required.
-  if clang-format --style=Google --output-replacements-xml "$filename" | grep -q "<replacement "; then
+  if ${CXX_CHECKER} --style=file --output-replacements-xml "$filename" | grep -q "<replacement "; then
```
Comment (95-martin-orion) [inline scripts/check_format.sh]: Why did we need to move away from `--style=Google`? Is this related to running in Docker?

Same for the YAPF check.

File: .clang-format [hunk 4]
```diff
@@ -0,0 +1,151 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Google
```
Comment (95-martin-orion) [inline .clang-format]: As I understand it, `BasedOnStyle` suggests that the file contains deviations from that style. Is that the case here, or is this a snapshot of the current Google style? Either way, could you clarify this in the comment?

File: .style.yapf [hunk 5]
```diff
@@ -0,0 +1,332 @@
+[style]
```
Comment (95-martin-orion) [inline .style.yapf]: Same as with `.clang-format` - could you add a comment on whether this is a snapshot of the current Google style?

File: .travis.yml (hunk 1 above)
Comment (s-mandra) [reply .travis.yml]: `alpine` doesn't have in their repo `yapf` but I can install it through `pip`. However, it makes sense to switch to `alpine`. Thanks!

File: .clang-format (hunk 4 above)
Comment (s-mandra) [reply .clang-format]: `.clang-format` should be the format file followed by `clang-format` for the Google style. I dumped it as:
```
clang-format --style google --dump-config > .clang-format
```
Other than that, I'm not familiar with `clang` format.

File: .style.yapf (hunk 5 above)
Comment (s-mandra) [reply .style.yapf]: Yes, it is by dumping the style as:
```
yapf3 --style google --style-help > .style.yapf
```

File: .travis.yml (hunk 1 above)
Comment (s-mandra) [reply .travis.yml]: Fixed.

File: scripts/check_format.sh (hunk 3 above)
Comment (s-mandra) [reply scripts/check_format.sh]: Different version of `yapf`/`clang-format` gives different results. Which means that if `yapf` or `clang-format` is just updated, it could fail the build. I've dumped the Google style for for both and put on `.clang-format` and `.style.yapf` (by default, `yapf` looks for this file) in the root folder. Also, I fixed the version of alpine to `3.10.3`. In this way, it is ensured that style remain consistent till we update the corresponding style files.

If you prefer, I can remove this limitation.

Comment (95-martin-orion) [reply scripts/check_format.sh]: This seems reasonable.

File: .style.yapf (hunk 5 above)
Comment (95-martin-orion) [reply .style.yapf]: Could you explain this in a comment in the file? Same for YAPF.

Comment (s-mandra) [reply .style.yapf]: Done, thanks!

File: scripts/check_format.sh (hunk 2 above)
Comment (s-mandra) [reply scripts/check_format.sh]: @alankao64 confirms (after installing `coreutils` using `brew`).
